src/wallet/gtest/test_rpc_wallet.cpp

```diff
@@ -0,0 +1,351 @@
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include "main.h"
+#include "primitives/transaction.h"
+#include "consensus/validation.h"
+#include "transaction_builder.h"
+#include "utiltest.h"
+#include "wallet/asyncrpcoperation_mergetoaddress.h"
+#include "wallet/asyncrpcoperation_shieldcoinbase.h"
+#include "wallet/asyncrpcoperation_sendmany.h"
+#include "zcash/JoinSplit.hpp"
+
+#include <librustzcash.h>
+#include <rust/ed25519.h>
+
+bool find_error(const UniValue& objError, const std::string& expected) {
+    return find_value(objError, "message").get_str().find(expected) != string::npos;
+}
+
+TEST(WalletRPCTests, RPCZsendmanyTaddrToSapling)
+{
+    LoadProofParameters();
+    SelectParams(CBaseChainParams::TESTNET);
+
+    LoadGlobalWallet();
+
+    RegtestActivateSapling();
+    {
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    if (!pwalletMain->HaveMnemonicSeed()) {
+        pwalletMain->GenerateNewSeed();
+    }
+
+    UniValue retValue;
+
+    KeyIO keyIO(Params());
+    // add keys manually
+    auto taddr = pwalletMain->GenerateNewKey(true).GetID();
+    auto pa = pwalletMain->GenerateNewLegacySaplingZKey();
+
+    const Consensus::Params& consensusParams = Params().GetConsensus();
+    retValue = chainActive.Height();
+    int nextBlockHeight = retValue.get_int() + 1;
+
+    // Add a fake transaction to the wallet
+    CMutableTransaction mtx = CreateNewContextualCMutableTransaction(consensusParams, nextBlockHeight);
+    CScript scriptPubKey = CScript() << OP_DUP << OP_HASH160 << ToByteVector(taddr) << OP_EQUALVERIFY << OP_CHECKSIG;
+    mtx.vout.push_back(CTxOut(5 * COIN, scriptPubKey));
+    CWalletTx wtx(pwalletMain, mtx);
+    pwalletMain->AddToWallet(wtx, true, NULL);
+
+    // Fake-mine the transaction
+    EXPECT_EQ(-1, chainActive.Height());
```

Comment:
This previously checked against the value 0 for `chainActive.Height()`, what was the reason for this change?

Reply:
i think i tried that and it reproducibly failed; and i looked at how other gtest tests did fake-mining and it started off at -1. i'm away from computer right now but i'll look at this when i get back. thanks for pointing that out; it confused me and i should have documented it more explicitly.

Reply:
Yeah, [all the fake-mining](https://github.com/zcash/zcash/blob/master/src/wallet/gtest/test_wallet.cpp#L232) (lines 232, 721, 831, 885, 972, 1102, 1931) in gtest's `test_wallet.cpp` starts off with a `EXPECT_EQ(-1, chainActive.Height());`, which i believe corresponds to "there's no blocks on the chain" -- so `Height()` returns `-1`.

`WalletTestingSetup`, the btest setup class for `rpc_z_sendmany_taddr_to_sapling`, inherits from `TestingSetup`, and i believe `TestingSetup` *does* give us a genesis block: `TestingSetup::TestingSetup` calls `InitBlockIndex`, which [implies that it will load a genesis block](https://github.com/zcash/zcash/blob/master/src/main.cpp#L5737) if we're not reindexing (and i don't see `fReindex` being set).

This is also consistent with the fact that the btest fake mining code does `chainActive.Tip()->GetBlockHash()`, which would segfault if there was no previous block (and none of the initial gtest fake-mines do that).